File: back-end/src/main/java/com/ta2khu75/quiz/service/AccountService.java
package com.ta2khu75.quiz.service;

import org.springframework.data.domain.Pageable;

import com.ta2khu75.quiz.model.request.AccountRequest;
import com.ta2khu75.quiz.model.request.update.AccountInfoRequest;
import com.ta2khu75.quiz.model.request.update.AccountStatusRequest;
import com.ta2khu75.quiz.model.response.AccountResponse;
import com.ta2khu75.quiz.model.response.PageResponse;
import com.ta2khu75.quiz.model.response.details.AccountAuthDetailsResponse;
import com.ta2khu75.quiz.model.response.details.AccountDetailsResponse;

public interface AccountService extends BaseService<AccountRequest, AccountResponse, String>{
    AccountAuthDetailsResponse updateStatus(String id,AccountStatusRequest request);
    PageResponse<AccountAuthDetailsResponse> readPage(String search,Pageable pageable);
    AccountResponse updateInfo(AccountInfoRequest request);
    AccountDetailsResponse readDetails(String id);
    AccountAuthDetailsResponse updateLock(String id);
}

File: front-end/src/service/AccountService.ts
import { AccountRequest } from "../model/request/AccountRequest";
import AccountPasswordRequest from "../model/request/update/AccountPasswordRequest";
import { AccountStatusRequest } from "../model/request/update/AccountStatusRequest";
import AccountResponse from "../model/response/AccountResponse";
import ApiResponse from "../model/response/ApiResponse";
import AccountDetailsResponse from "../model/response/details/AccountDetailsResponse";
import PageResponse from "../model/response/PageResponse";
import instance from "../util/apiInstance";

const basePath = "account";
export default class AccountService {
  static readPage(search = "", page = 0, size = 10): Promise<ApiResponse<PageResponse<AccountDetailsResponse>>> {
    return instance.get(`${basePath}`, { params: { search, page, size } });
  }
  static create(
    account: AccountRequest
  ): Promise<ApiResponse<AccountResponse>> {
    return instance.post(basePath, account);
  }
  static updateStatus(id: string, account: AccountStatusRequest): Promise<ApiResponse<AccountDetailsResponse>> {
    return instance.put(`${basePath}/${id}`, account);
  }

  static updatePermission(id: string, permissionIds:number[]): Promise<ApiResponse<AccountDetailsResponse>> {
    return instance.put(`${basePath}/${id}/permission`, { permission_ids: permissionIds });
  }
  static readById(id: number): Promise<ApiResponse<AccountResponse>> {
    return instance.get(`${basePath}/${id}`);
  }
  static changePassword(
    accountPassword: AccountPasswordRequest
  ): Promise<ApiResponse<AccountPasswordRequest>> {
    return instance.put(`${basePath}/change-password`, accountPassword);
  }
}
